### PR TITLE
feat: Add event tracker  [DEV-3530]

### PR DESCRIPTION
**Pull request by lampkin-diet**
- Add the observer pattern implementation for tracking event handling
- For now, there is only one event to track - `track` but it includes some operations inside.
- By default enables logging and `operation` table filling
- Adds the abstract handler for datadog without explicit implementation

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-3530 Alerting/monitoring feature for tracking operation failures](https://app.clickup.com/t/6600954/DEV-3530)

File: src/controllers/credential-status.ts [hunk 1]
```diff
@@ -496,6 +498,12 @@ export class CredentialStatusController {
 
 		// handle error
 		if (!result.isEmpty()) {
+			const message = result.array().pop()?.msg;
+			eventTracker.emit('notify', {
+				message: `createUnencryptedStatusList. Request validation error: ${message}`,
+				severity: 'error',
```
Comment (Eengineer1) [inline src/controllers/credential-status.ts:504]: Validation errors are user-originated, which is not an error per se.

We should probably lower the severity to `info`.

File: src/controllers/credential-status.ts [hunk 2]
```diff
@@ -544,35 +552,35 @@ export class CredentialStatusController {
 
 			// handle error
 			if (result.error) {
+				const message = result.error?.message || result.error.toString();
+				eventTracker.emit('notify', {
+					message: `createUnencryptedStatusList. Request validation error: ${message}`,
+					severity: 'error',
```
Comment (Eengineer1) [inline src/controllers/credential-status.ts:558]: Marking to assess by lowering severity.

File: src/controllers/credentials.ts [hunk 3]
```diff
@@ -6,9 +6,8 @@ import { check, validationResult, query } from './validator/index.js';
 
 import { Credentials } from '../services/credentials.js';
 import { IdentityServiceStrategySetup } from '../services/identity/index.js';
-import type { ITrackOperation, ValidationErrorResponseBody } from '../types/shared.js';
-import { Cheqd } from '@cheqd/did-provider-cheqd';
-import { OPERATION_CATEGORY_NAME_CREDENTIAL } from '../types/constants.js';
+import type { ValidationErrorResponseBody } from '../types/shared.js';
+// import { Cheqd } from '@cheqd/did-provider-cheqd';
```
Comment (Eengineer1) [inline src/controllers/credentials.ts:10]: Marking to remove.

File: src/controllers/presentation.ts [hunk 4]
```diff
@@ -199,6 +202,7 @@ export class PresentationController {
 		const cheqdPresentation = new CheqdW3CVerifiablePresentation(presentation);
 
 		if (makeFeePayment) {
+			// ToDo: Add Payments tracker
```
Comment (Eengineer1) [inline src/controllers/presentation.ts:205]: Since this is in scope of this PR, let's implement + include in Credential Status operations as well.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
The core work is very solid.

There are certain points to address, before that's good to go.

File: src/controllers/presentation.ts (hunk 4 above)
Comment (lampkin-diet) [reply src/controllers/presentation.ts:205]: Hey! Such payment tracking feature has been implementing in the PR with minor tracking things.

**Review by Eengineer1 — APPROVED** [review]

**Pull request merged**